nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkServiceImpl.java

```diff
@@ -0,0 +1,154 @@
+        if (log.isDebugEnabled()) {
+            log.debug("Run operation with command=" + command);
+        }
+        // fill command object
+        if (isEmpty(command.getRepository())) {
+            String repository = Framework.getService(RepositoryManager.class).getDefaultRepositoryName();
+            command.withRepository(repository);
+        }
+        // create the operation id and status
+        UUID bulkOperationId = UUID.randomUUID();
+
+        BulkStatus status = new BulkStatus();
+        status.setUUID(bulkOperationId);
+        status.setState(SCHEDULED);
+        status.setCreationDate(ZonedDateTime.now());
+        status.setCommand(command);
+
+        try {
+            byte[] commandAsBytes = OBJECT_MAPPER.writeValueAsBytes(command);
+
+            // store the bulk command and status in the key/value store
+            KeyValueStore keyValueStore = getKvStore();
+            keyValueStore.put(bulkOperationId + STATE, status.getState().toString());
+            keyValueStore.put(bulkOperationId + CREATION_DATE, status.getCreationDate().toString());
+            keyValueStore.put(bulkOperationId + COMMAND, commandAsBytes);
+
+            // send it to nuxeo-stream
+            String key = bulkOperationId.toString();
+            getLogManager().getAppender(SET_STREAM_NAME).append(key, new Record(key, commandAsBytes));
+        } catch (JsonProcessingException e) {
+            throw new NuxeoException("Unable to serialize the bulk command=" + command, e);
+        }
+        return status;
+    }
+
+    @Override
+    public BulkStatus getStatus(UUID bulkOperationId) {
+        String commandAsString = "";
+        try {
+            BulkStatus status = new BulkStatus();
+            status.setUUID(bulkOperationId);
+
+            // retrieve values from KeyValueStore
+            KeyValueStore keyValueStore = getKvStore();
+            String state = keyValueStore.getString(bulkOperationId + STATE);
+            status.setState(State.valueOf(state));
+
+            String creationDate = keyValueStore.getString(bulkOperationId + CREATION_DATE);
+            status.setCreationDate(ZonedDateTime.parse(creationDate));
+
+            commandAsString = keyValueStore.getString(bulkOperationId + COMMAND);
+            BulkCommand command = OBJECT_MAPPER.readValue(commandAsString, BulkCommand.class);
+            status.setCommand(command);
+
+            Long scrolledDocumentCount = keyValueStore.getLong(bulkOperationId + SCROLLED_DOCUMENT_COUNT);
+            status.setScrolledDocumentCount(scrolledDocumentCount);
+
+            return status;
+        } catch (IOException e) {
+            throw new NuxeoException("Unable to deserialize the bulk command=" + commandAsString, e);
+        }
+    }
+
+    public KeyValueStore getKvStore() {
+        if (kvStore == null) {
+            kvStore = Framework.getService(KeyValueService.class).getKeyValueStore(descriptor.kvStore);
```

Comment:
Do we really need such a local cache system, is it expected to be called a lot? I'd simply call the service when needed.